homeassistant/components/renson/button.py

```diff
@@ -0,0 +1,73 @@
+"""Renson ventilation unit buttons."""
+from __future__ import annotations
+
+from _collections_abc import Callable
+from renson_endura_delta.renson import RensonVentilation
+
+from homeassistant.components.button import (
+    ButtonDeviceClass,
+    ButtonEntity,
+    ButtonEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import RensonCoordinator, RensonData
+from .const import DOMAIN
+from .entity import RensonEntity
+
+SYNC_TIME_BUTTON: ButtonEntityDescription = ButtonEntityDescription(
+    key="sync_time",
+    entity_category=EntityCategory.CONFIG,
+    translation_key="sync_time",
+)
+
+RESTART_BUTTON: ButtonEntityDescription = ButtonEntityDescription(
+    key="restart",
+    device_class=ButtonDeviceClass.RESTART,
+    entity_category=EntityCategory.CONFIG,
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the Renson button platform."""
+
+    data: RensonData = hass.data[DOMAIN][config_entry.entry_id]
+
+    entities = [
+        RensonButton(SYNC_TIME_BUTTON, data.api, data.coordinator, data.api.sync_time),
+        RensonButton(
+            RESTART_BUTTON, data.api, data.coordinator, data.api.restart_device
+        ),
+    ]
+
+    async_add_entities(entities)
+
+
+class RensonButton(RensonEntity, ButtonEntity):
+    """Representation of a Renson actions."""
+
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        description: ButtonEntityDescription,
+        api: RensonVentilation,
+        coordinator: RensonCoordinator,
+        action: Callable,
+    ) -> None:
+        """Initialize class."""
+        super().__init__(description.key, api, coordinator)
+
+        self.entity_description = description
+        self._action = action
+
+    async def async_press(self) -> None:
+        """Triggers the action."""
+        await self.hass.async_add_executor_job(self._action)
```

Comment:
```suggestion
        await self.hass.async_add_executor_job(
            self.entity_description.action_fn(self.api)
        )
```

Reply:
This code gives some error:
RuntimeError: Blocking calls must be done in the executor or a separate thread; Use `await hass.async_add_executor_job()`; at homeassistant/components/renson/button.py, line 41: action_fn=lambda api: api.sync_time(),

I'm trying to fix this but cannot find it at the moment